crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -40,31 +40,54 @@ def supported?
 
     def knife_files
       @data.join("knife", "databags", "barclamps").rmtree
+      knife_path = @data.join("knife")
 
-      crowbar_databags_path = @data.join("knife", "databags", "crowbar")
+      crowbar_databags_path = knife_path.join("databags", "crowbar")
       crowbar_databags_path.children.each do |file|
-        if file == "bc-template-nova_dashboard.json"
-          crowbar_databags_path.join(file).rename(
-            crowbar_databags_path.join(file.sub!("nova_dashboard", "horizon"))
-          )
+        file_path = crowbar_databags_path.join(file)
 
-          file_content = File.read(crowbar_databags_path.join(file))
-          file_content.gsub!("nova_dashboard", "horizon")
-          File.open(crowbar_databags_path.join(file), "w") { |content| content.puts file_content }
+        if file == "bc-template-nova_dashboard.json"
+          new_file = rename_file(file_path, "nova_dashboard", "horizon")
+          replace_file_content(new_file, "nova_dashboard", "horizon")
+          file_path = new_file
         end
 
-        next unless file.to_s.match("bc-.*.json")
-        new_file = file.sub("bc-", "")
-        crowbar_databags_path.join(file).rename(new_file)
+        next unless file_path.to_s.match("bc-.*.json")
+        file_path = rename_file(file_path, "bc-", "")
+        replace_file_content(file_path, "bc-", "")
+      end
 
-        file_content = crowbar_databags_path.join(new_file).read
-        file_content.gsub!("bc-", "")
-        File.open(crowbar_databags_path.join(new_file), "w") { |content| content.puts file_content }
+      barclamps_databag_path = knife_path.join("barclamps")
+      barclamps_databag_path.children.each do |file|
+        case file.to_s
+        when "nova_dashboard.json"
+          new_file = rename_file(file, "nova_dashboard", "horizon")
+          replace_file_content(new_file, "nova_dashboard", "horizon")
+        when "tempest.json" || "openstack.json"
+          replace_file_content(new_file, "nova_dashboard", "horizon")
+        end
       end
+
+      roles_path = knife_path.join("roles")
+      file = roles_path.join("nova_dashboard-server.json")
+      new_file = rename_file(file, "nova_dashboard", "horizon")
+      replace_file_content(new_file, "nova_dashboard", "horizon")
     end
 
     def crowbar_files
       FileUtils.touch(@data.join("crowbar", "production.yml"))
     end
+
+    def rename_file(file, search, replace)
```

Comment:
The name is not exactly what the method does, it's a rename with a regex. So this should be reflected in the name and then we can move this to our utils lib.